vars/sonarExecuteScan.groovy

```diff
@@ -110,14 +119,20 @@ void call(Map parameters = [:]) {
 
         def worker = { config ->
             withSonarQubeEnv(config.instance) {
-                loadSonarScanner(config)
+                try{
+                    loadSonarScanner(config)
+
+                    loadCertificates(config)
 
-                if(config.organization) config.options.add("sonar.organization=${config.organization}")
-                if(config.projectVersion) config.options.add("sonar.projectVersion=${config.projectVersion}")
-                // prefix options
-                config.options = config.options.collect { it.startsWith('-D') ? it : "-D${it}" }
+                    if(config.organization) config.options.add("sonar.organization=${config.organization}")
+                    if(config.projectVersion) config.options.add("sonar.projectVersion=${config.projectVersion}")
+                    // prefix options
+                    config.options = config.options.collect { it.startsWith('-D') ? it : "-D${it}" }
 
-                sh "PATH=\$PATH:${env.WORKSPACE}/.sonar-scanner/bin sonar-scanner ${config.options.join(' ')}"
+                    sh "PATH=\$PATH:${env.WORKSPACE}/.sonar-scanner/bin sonar-scanner ${config.options.join(' ')}"
+                }finally{
+                    def ignore = sh script: "rm -rf .sonar-scanner .certificates .scannerwork", returnStatus: true
```

Comment:
* Unusual pattern: Assign something to a variable named ignore and do nothing with it. Why assigning it at all?
* Is it important not to have the certificates on the file system afterwards?
  * In this case checking the return code would be good [1]? 
  * The certificates are removed from file system. Do we need to remove the certificates also from the keystore? In case is is only some kind of cleanup ( ... and not for security reasons) we could also remove the certificates immediately after adding to the keystore.
  * In case an exception is raised before handling the certificates we end up in the try block and we cannot remove the certificates folder since it does not exist.

Footnotes
[1] with `-f` flag on `rm` we do not get an exit code !=0 in case a file does not exist.

Reply:
> Is it important not to have the certificates on the file system afterwards?

It is just a safety task to ensure all created files are removed afterwards as the container runs with user `root`

> Do we need to remove the certificates also from the keystore?

No, the `keystore` is part of the container which is deleted afterwards. The Workspace is mounted into the container.

Reply:
> [1] with -f flag on rm we do not get an exit code !=0 in case a file does not exist.
 
Actually I don't want to handle missing files.

Reply:
statusCode handling is removed